Description: Allow disabling PSF plugin.
Author: John Lindgren <john.lindgren@tds.net>
Bug: http://jira.atheme.org/browse/AUDPLUG-290
Origin: backport, http://hg.atheme.org/audacious-plugins/audacious-plugins/rev/71d049eb57b4

--- a/configure.ac
+++ b/configure.ac
@@ -127,7 +127,7 @@
 
 dnl These plugins are always built.
 
-INPUT_PLUGINS="tonegen console psf xsf metronom vtx"
+INPUT_PLUGINS="tonegen console xsf metronom vtx"
 OUTPUT_PLUGINS="null"
 EFFECT_PLUGINS="compressor crossfade crystalizer ladspa sndstretch stereo_plugin voice_removal echo_plugin"
 GENERAL_PLUGINS="song_change alarm skins gtkui lyricwiki"
@@ -356,6 +356,20 @@
 AC_SUBST(SIMD_CFLAGS)
 CFLAGS="$ac_save_CFLAGS"
 
+
+dnl PSF
+dnl ===
+
+AC_ARG_ENABLE(psf,
+ [  --disable-psf            disable PlayStation (psf/psf2) audio decoder],
+ [enable_psf=$enableval],
+ [enable_psf=yes])
+
+if test "$enable_psf" = "yes"; then
+    INPUT_PLUGINS="$INPUT_PLUGINS psf"
+fi
+
+
 dnl *** USF
 
 AC_ARG_ENABLE(usf,
@@ -1636,7 +1650,7 @@
 echo "    -> libSIDPlay2 support:               $xs_have_sidplay2"
 echo "    -> distortion patched libSIDPlay2:    $xs_have_distortion"
 echo "  Game music (spc, nsf & gbs):            yes"
-echo "  PlayStation (psf/psf2) audio (psf):     yes"
+echo "  PlayStation (psf/psf2) audio:           $enable_psf"
 echo "  Nintendo 64 audio (usf):                $enable_usf"
 echo "  Nintendo DS audio (xsf):                yes [experimental]"
 echo "  AdLib synthesizer (adplug):             $have_adplug"
